src/main/java/org/apache/maven/shared/filtering/DefaultMavenFileFilter.java

```diff
@@ -105,8 +110,6 @@ public void copyFile( File from, File to, boolean filtering, List<FileUtils.Filt
                 }
                 FileUtils.copyFile( from, to, encoding, new FileUtils.FilterWrapper[0], overwrite );
             }
-
-            buildContext.refresh( to );
```

Comment:
isn't something used by m2e? or not anymore?

Reply:
This comes from a repo that is last touched 12 years ago, and is archived. IMHO is dead, parrot dead, but we can tinker about it if you want :smile:

Reply:
a quick search on github shows this https://github.com/search?q=org%3Aeclipse-m2e+BuildContext&type=code
for sure this very old repo is just interface which doesn't need much changes so that's probably the reason.

disclaimer i'm not using m2e so... 🤣

Reply:
You are missing the point: this parrot is dead. Very. Also, whatever it is, this was first attempt to implement incremental build, followed by two other attempts (by same person), that finally materialized in something completely different. So, yes, this may look "stable", the very same way dead parrot looks "stable" (well, not moving, is it?), but is superseded by several other attempts...

Reply:
maybe but I can see references of it in the m2e code.
Is it really used or not? I have no idea as I'm not a contributor of m2e.
My opinion is just it worth to ask those folks as we may or not break something.
if not perfect but it worth the courtesy of simply asking.

Reply:
Frankly, this should be other way around, don't you agree? This dead parrot (not authored by us, not used by anything other than m2e) does not buy anything for Maven itself (just adds extra complexity and more dependencies), does not buy anything for any other IDE out there either and finally does not buy anything for end users either. There is one and only one "beneficiary" of this code, if any.

In other words, if tomorrow some Big Team approaches us, wanting to put some "custom code" into Maven Core or Plugins to make it work for them, what would you say? And next day another team, and so on... 

Anyway, we are trying to move forward, spring cleanup time (pending since a lot years), and we do accept quality patches as well, so nothing is lost here IMO by removing this.

Reply:
fyi there are still some interest here https://github.com/codehaus-plexus/plexus-build-api/issues

Reply:
heh, so yet another project that is superseded but is revived and babysit? Well, the problem is as I wrote above: this project is dead, deal with it. The logical successor is https://github.com/takari/io.takari.incrementalbuild

Reply:
frankly looking at commit history m2e has more maintenance than https://github.com/takari/io.takari.incrementalbuild (last commit almost 2 yo ago).

Reply:
And what does it has to do with "liveliness" of 12 years dead project? Are you mixing up things maybe? sonatype/sisu-build-api is NOT m2e, so unsure what m2e "activity" has to do here. Also, you commented above how this API is "stable" (so stable that it did not need change for 12 years), and now you rant about measly 2 years?  :smile:

Reply:
If I get it correctly, what you say is "buildcontext prevails over takari/incrementalbuild as m2e has more commits"? :kangaroo: LOL, well, I am about to stop here.... :laughing:

Reply:
> If I get it correctly, what you say is "buildcontext prevails over takari/incrementalbuild as m2e has more commits"? 🦘 LOL, well, I am about to stop here.... 😆

no sorry but you do not read correctly, the most important is the number of users impacted.
and here in m2e is a lot!
what is better it's not the question here. 
again I'm just saying the impacted users number is really huge!

Reply:
> frankly looking at commit history m2e has more maintenance than https://github.com/takari/io.takari.incrementalbuild (last commit almost 2 yo ago).

And who said this then? 

Anyway, lets bury the boomerang, sisu build api is raised from the dead

Reply:
... and my 5 cents to whole this topic: I never saw a "plain" (explained later) Maven project that is fairly more complex than "hello world" and works OOTB in m2e. Work, as in no "tweaks", "m2e profile in POM" needed etc. OTOH, I did saw Maven project using takari lifecycle that WORK perfectly and OOTB in m2e. 

But alas, takari lifecycle use the "inferior" incrementalbuild and not buildcontext API. Oh, and it also replaces many things from Maven itself, and even ditches this maven-filtering and replaces it's with it's own to be functional :smile: 

("plain" as Maven build w/o any extension, while takari-lifecycle is an extension and replaces a LOT from maven, like core copmponets but also shared stuff like this maven-filtering and many plugins as well)